TestResultSummaryService/parsers/BenchmarkParser.js

```diff
@@ -100,21 +100,27 @@ class BenchmarkParser extends Parser {
                 
                 // Parse metric values
                 curTestData["metrics"] = [];
-                for ( let i = 0; i < curBenchVariant["metrics"].length; i++ ) {
+                let curMetricList = Object.keys(curBenchVariant["metrics"]);
+                for ( let i = 0; i < curMetricList.length; i++ ) {
 
-                    curMetric = curBenchVariant["metrics"][i];  
+                    curMetric = curMetricList[i];
+                    curRegex = curBenchVariant["metrics"][curMetric]["regex"];  
                     /*	Parse all values for single metric from result to an array
                      *  e.g 
                      *  Liberty Startup =>
                      *  curRegexResult = ['startup time in ms 32',32,'startup time in ms 41',41, x 6] 
                      *  Liberty Throughput =>
                      *  curRegexResult = ['<metric type="throughput">32<\/data>',32]
                      */
-                    curRegexResult = curSearchString.split(curMetric.regex);
+                    curRegexResult = curSearchString.split(curRegex);
                     //collect only the capture groups from the regex array
-                    curRegexResult = curRegexResult.filter ( (value,index) => (index %2 == 1) );
+                    curRegexResult = curRegexResult.filter( (value,index) => (index %2 == 1));
                     curMetricValue = curRegexResult.map(parseFloat);
```

Comment:
Let's rename `curMetricValue` to `curMeticValues` since it can store multiple values and not just one.